.Lib9c.Tests/Action/ValidatorDelegation/GasWithTransferAssetTest.cs

```diff
@@ -0,0 +1,205 @@
+        EnsureToMintAsset(signerKey, NCG * 100);
+
+        // When
+        var amount = NCG * 1;
+        var expectedMead = Mead * 0;
+        var expectedNCG = NCG * 0;
+        var transferAsset = new TransferAsset(
+            signerKey.Address, recipientKey.Address, amount, memo: "test");
+        MakeTransaction(
+            signerKey,
+            new ActionBase[] { transferAsset, },
+            maxGasPrice: Mead * 1,
+            gasLimit: gasLimit);
+
+        // Then
+        var e = Assert.Throws<AggregateException>(() => MoveToNextBlock(throwOnError: true));
+        var innerExceptions = e.InnerExceptions
+            .Cast<UnexpectedlyTerminatedActionException>()
+            .ToArray();
+        var actualNCG = GetBalance(recipientKey.Address, NCG);
+        var actualMead = GetBalance(signerKey.Address, Mead);
+        Assert.Single(innerExceptions);
+        Assert.Contains(innerExceptions, i => i.Action is TransferAsset);
+        Assert.Equal(expectedNCG, actualNCG);
+        Assert.Equal(expectedMead, actualMead);
+    }
+
+    [Theory]
+    [InlineData(3, 5)]
+    [InlineData(1, 10)]
+    public void Execute_ExcceedGasLimit_Throw(
+        long gasLimit, long gasOwned)
+    {
+        if (gasLimit >= GasConsumption)
+        {
+            throw new ArgumentOutOfRangeException(
+                nameof(gasLimit),
+                $"{nameof(gasLimit)} must be less than {GasConsumption}.");
+        }
+
+        if (gasOwned < GasConsumption)
+        {
+            throw new ArgumentOutOfRangeException(
+                nameof(gasOwned),
+                $"{nameof(gasOwned)} must be greater than or equal to {GasConsumption}.");
+        }
+
+        // Given
+        var amount = NCG * 1;
+        var signerKey = new PrivateKey();
+        var signerMead = Mead * gasOwned;
+        var recipientKey = new PrivateKey();
+        EnsureToMintAsset(signerKey, signerMead);
+        EnsureToMintAsset(signerKey, NCG * 100);
+
+        // When
+        var expectedMead = signerMead - (Mead * gasLimit);
+        var expectedNCG = NCG * 0;
+        var transferAsset = new TransferAsset(
+            signerKey.Address, recipientKey.Address, amount, memo: "test");
+        MakeTransaction(
+            signerKey,
+            new ActionBase[] { transferAsset, },
+            maxGasPrice: Mead * 1,
+            gasLimit: gasLimit);
+
+        // Then
+        var e = Assert.Throws<AggregateException>(() => MoveToNextBlock(throwOnError: true));
+        var innerExceptions = e.InnerExceptions
+            .Cast<UnexpectedlyTerminatedActionException>()
+            .ToArray();
+        var actualNCG = GetBalance(recipientKey.Address, NCG);
+        var actualMead = GetBalance(signerKey.Address, Mead);
+        Assert.Single(innerExceptions);
+        Assert.Contains(innerExceptions, i => i.Action is TransferAsset);
+        Assert.Equal(expectedNCG, actualNCG);
+        Assert.Equal(expectedMead, actualMead);
+    }
+}
```

Comment:
Do we need to have both of `GasTest` and `GasWithTransferAssetTest`? At a glance, they seems like similar.

Reply:
> Do we need to have both of `GasTest` and `GasWithTransferAssetTest`? At a glance, they seems like similar.

The purpose of this test code is to verify that actions used in lib9c function properly for gas collection. I wanted to create a more complex test, but since only one action is allowed per transaction, I couldn't achieve that. It ended up looking like GasTest, inevitably. :)